### ISSUE TITLE
Let's use `getent ahosts` instead of `dig +short`

### ISSUE THREAD

**Issue by vazhnov** [mmumshad/kubernetes-the-hard-way#355]
### Your Workstation

Other (specify in What Happened box)

### System Memory

More than 8GB, less than 16GB

### What happened?

Hello!

Thank you for the project!

I recommend to use `getent ahosts NAME` instead of `dig +short NAME` in this course, for both: manual steps in the documentation and scripts.

`getent` is a tool usually included into even very minimalistic OS setup.

`getent` follows `/etc/nsswitch.conf` rules (reads `/etc/hosts`, mDN·S, etc.), while `dig` is mainly designed to troubleshoot external DNS servers. Much more details about the difference can be found in https://unix.stackexchange.com/questions/20784/ (or in my [brief summary](https://vazhnov.gitlab.io/Cheat_sheet_Admin_public.html#a__DNS_shell)).

I think using `dig` is a bad practice when custom records in `/etc/hosts` are used. `getent` is designed for such cases.

I'm currently in step [08](https://github.com/mmumshad/kubernetes-the-hard-way/blob/master/docs/08-bootstrapping-kubernetes-controllers.md) and I see no issues with this replacement.

Also I've checked: using `getent ahosts` is compatible with both Vagrant images, Ubuntu and Debian, to do steps of this course.

<del>Currently script [vagrant/ubuntu/update-dns.sh](https://github.com/mmumshad/kubernetes-the-hard-way/blob/master/vagrant/ubuntu/update-dns.sh) fails to apply in Debian</del> let's skip this for dedicated issue.

**Comment by fireflycons** [comment]
I'm using `dig +short` because it returns *only* the IP address with no need to muck about with `awk` to get it. `getent` will return IP+HOST
The lab's not millisecond performance critical so if dig does waste time getting a negative response from DNS before checking hosts file, not really an issue.

**Comment by vazhnov** [comment]
> I'm using dig +short because it returns only the IP address with no need to muck about with awk to get it. getent will return IP+HOST

I agree, I have to use `getent ahosts example.com | awk '{ print $1 ; exit }'` to return only one IP-address without name.

> The lab's not millisecond performance critical so if dig does waste time getting a negative response from DNS before checking hosts file, not really an issue.

The thing is — `dig` doesn't use `/etc/hosts` at all. In case of the lab, I think that systemd-resolved is doing that.

This change is also about to teach people to use proper tools.
